packages/lesswrong/lib/analyticsEvents/ultraFeedEvents.ts

```diff
@@ -0,0 +1,130 @@
+import { useRef, useEffect, useCallback } from 'react';
+import { useMutation } from '@apollo/client';
+import gql from 'graphql-tag';
+
+interface UltraFeedEventInput {
+  eventType: 'viewed' | 'expanded';
+  documentId: string;
+  documentType: 'post' | 'comment' | 'spotlight';
+  postId?: string;
+  expansionLevel?: number;
+  maxExpansionReached?: boolean;
+  wordCount?: number;
+}
+
+const LOG_ULTRA_FEED_EVENT = gql`
```

Comment:
This should be defined inside of the hook for performance reasons (since otherwise importing this file incurs the cost of parsing, which slows down client-side page loading)

Reply:
deleted anyway